File: ecommerce-frontend/src/services/AuthService.js
import api from "../api/api";

const login = (data) => {
    return api.post("/auth/login", data);
};

const register = (data) => {
    return api.post("/auth/register", data);
};

const getCurrentUser = () => {

    return api.get("/auth/me");

};

export default {

    register,

    login,

    getCurrentUser

};

File: ecommerce-backend/src/main/java/com/ecommerce/service/AuthService.java
package com.ecommerce.service;

import com.ecommerce.dto.LoginRequest;
import com.ecommerce.dto.LoginResponse;
import com.ecommerce.dto.RegisterRequest;
import com.ecommerce.dto.UserResponse;

public interface AuthService {

    String register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    UserResponse getCurrentUser();

}